test/e2e/quick_start_test.go

```diff
@@ -36,63 +40,137 @@ var (
 	}
 )
 
-var _ = Describe("Docker examples [Docker]", Serial, func() {
-	for _, specName := range []string{
-		"topology-cilium-helm-addon [HelmAddon][Cilium]",
-		"topology-cilium-crs [ClusterResourceSet][Cilium]",
-		"topology-calico-helm-addon [HelmAddon][Calico]",
-		"topology-calico-crs [ClusterResourceSet][Calico]",
-	} {
-		flavour, _, found := strings.Cut(specName, " ")
-		Expect(found).To(BeTrue(), "template has invalid format")
-		Context(specName, func() {
-			e2e.QuickStartSpec(ctx, func() e2e.QuickStartSpecInput {
-				return e2e.QuickStartSpecInput{
-					E2EConfig:              e2eConfig,
-					ClusterctlConfigPath:   clusterctlConfigPath,
-					BootstrapClusterProxy:  bootstrapClusterProxy,
-					ArtifactFolder:         artifactFolder,
-					SkipCleanup:            skipCleanup,
-					Flavor:                 ptr.To(flavour),
-					InfrastructureProvider: ptr.To("docker"),
-					PostMachinesProvisioned: func(proxy framework.ClusterProxy, namespace, clusterName string) {
-						framework.AssertOwnerReferences(namespace, proxy.GetKubeconfigPath(),
-							framework.CoreOwnerReferenceAssertion,
-							framework.ExpOwnerReferenceAssertions,
-							framework.DockerInfraOwnerReferenceAssertions,
-							framework.KubeadmBootstrapOwnerReferenceAssertions,
-							framework.KubeadmControlPlaneOwnerReferenceAssertions,
-							framework.KubernetesReferenceAssertions,
-							map[string]func(reference []metav1.OwnerReference) error{
-								clusterResourceSetKind: func(owners []metav1.OwnerReference) error {
-									// The ClusterResourcesSets that we create are cluster specific and so should be owned by the cluster.
-									return framework.HasExactOwners(owners, clusterOwner)
-								},
+var _ = Describe("Quick start", Serial, func() {
+	for _, provider := range []string{"Docker"} {
+		for _, cniProvider := range []string{"Cilium", "Calico"} {
+			for _, addonStrategy := range []string{"HelmAddon", "ClusterResourceSet"} {
+				strategy := ""
+				switch addonStrategy {
+				case "HelmAddon":
+					strategy = "helm-addon"
+				case "ClusterResourceSet":
+					strategy = "crs"
```

Comment:
Why is this needed but no special handling for `cniProvider`?

Reply:
Because the `Label` doesn't match the component in the cluster template name for strategy but does for CNI provider.

Cluster template name format is currently `topology-(cilium|calico)-(crs|helm-addon)` so for CNI provider component we can just lowercase the provider name, but have to modify the strategy as done here. It's a bit rubbish I guess but it provides for nice labels imo